### DISCUSSION TITLE
primary key in budibase db

### DISCUSSION THREAD

**Discussion by BellBoy68** [Budibase/budibase#10078] [Help] [answered]
Hello. Is it possible to make data in budibase db a primary key? If yes, how to do it?
i want the data in my budibase db not to repeat

**Comment by kasgithub8** [comment]
Did you create the Budibase table with the Auto ID? 
If so, that's basically your primary key.

**Reply by BellBoy68** [reply]
Then how do I check for duplicate data in the created column?

**Reply by kasgithub8** [reply]
What "created" column?

**Reply by BellBoy68** [reply]
I created a column with phone numbers and other columns. I would like to make sure that I do not have data with the same phone numbers.

**Reply by kasgithub8** [reply]
I see.
As far as my Budibase knowledge goes, that's not possible with the Budibase database.
If possible, you could try switching to MySQL.

**Reply by BellBoy68** [reply]
Ok. Thanks for the answer

**Comment by ConorWebb96** [comment, marked as answer]
Hey @BellBoy68,

What @kasgithub8 said is right. Just going to follow up with a few things which may help you.

There are a few things you could try in addition to what they said to find duplicates in your data with the Budibase db.

- You could ascend and descend the columns by what you are checking generally duplicates will show below one another.
- Additionally you can use the filter at the top of the data screen, if you have a specific number in mind paste it in the values area and select the field you wish to filter by. Once this has been added and saved it will filter your tables data by anything which matches it. (Some screenshots have been provided below to help you find certain features talked about above.)

**Asc/Desc Toggle**
<img width="133" alt="Screenshot 2023-03-20 at 12 16 47" src="https://user-images.githubusercontent.com/126772285/226337954-cd5bc999-1c17-4d5c-81f3-8ad71f461229.png">

**Filter on the far right**
<img width="536" alt="Screenshot 2023-03-20 at 12 18 07" src="https://user-images.githubusercontent.com/126772285/226337960-6c0e95fa-438c-47ff-8466-080d20523a20.png">

**Filter modal**
<img width="1204" alt="Screenshot 2023-03-20 at 12 18 24" src="https://user-images.githubusercontent.com/126772285/226337964-6050842a-f0ab-43ab-a514-2e4a544e7416.png">

**Filter output**
<img width="728" alt="Screenshot 2023-03-20 at 12 18 32" src="https://user-images.githubusercontent.com/126772285/226337972-231d4ce9-a0e8-4257-812f-68b059d5a3da.png">

Hopefully this helps!

**Reply by kasgithub8** [reply]
@ConorWebb96 Oh, I thought he tried to customize the fields so that you could not enter the same value twice.

**Reply by ConorWebb96** [reply]
I think they just wanted to see if there were any duplicates in the table for numbers from reading your response. I may be wrong though!

**Reply by BellBoy68** [reply]
> @ConorWebb96О, я думал, он пытался настроить поля так, чтобы нельзя было ввести одно и то же значение дважды.

That's exactly what I wanted

**Reply by kasgithub8** [reply]
> I think they just wanted to see if there were any duplicates in the table for numbers from reading your response. I may be wrong though!

You're right! 

@BellBoy68 my bad. 😭
